app/code/Magento/CatalogInventory/Model/StockStateProvider.php

```diff
@@ -106,6 +106,10 @@ public function checkQuoteItemQty(StockItemInterface $stockItem, $qty, $summaryQ
     {
         $result = $this->objectFactory->create();
         $result->setHasError(false);
+        
+        $product = $this->productFactory->create();
+        $product->load($stockItem->getProductId());
```

Comment:
Why not use `ProductRepository` here? Is product name always missing or only in some cases?